k2vr-installer-gui/Tools/Utils.cs

```diff
@@ -20,40 +20,57 @@ public static bool IsDirectoryEmpty(string path)
         public static bool EnsureSteamVrClosed()
         {
             Logger.Log("Checking if SteamVR is open...", false);
-            foreach (Process process in Process.GetProcesses())
+
+            if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
             {
-                if (process.ProcessName == "vrmonitor")
+                //Ask if SteamVR should be closed
+                if (MessageBox.Show(
+                    "SteamVR needs to be closed to continue the installation." + Environment.NewLine + "Do you want Setup to close SteamVR?",
+                    "Close SteamVR?",
+                    MessageBoxButton.YesNo,
+                    MessageBoxImage.Warning
+                    ) != MessageBoxResult.Yes)
                 {
-                    Logger.Log("Closing vrmonitor...", false);
-                    process.CloseMainWindow();
-                    Thread.Sleep(5000);
-                    if (!process.HasExited)
+                    //Only exit if SteamVR is still running
+                    if (Process.GetProcesses().FirstOrDefault((Process proc) => proc.ProcessName == "vrserver" || proc.ProcessName == "vrserver") != null)
                     {
-                        Logger.Log("Force closing...", false);
-                        /* When SteamVR is open with no headset detected,
-                           CloseMainWindow will only close the "headset not found" popup
-                           so we kill it, if it's still open */
-                        process.Kill();
-                        Thread.Sleep(3000);
+                        return false;
                     }
                 }
             }
 
+            //Close VrMonitor
+            foreach (Process process in Process.GetProcesses().Where((proc) => proc.ProcessName == "vrmonitor"))
+            {
+                Logger.Log("Closing vrmonitor (PID: " + process.Id + ")...", false);
```

Comment:
Is the PID really that relevant? (logs get displayed on the install screen, does the user care about the PID?)

Reply:
I though that it would be good to know the PID in case something goes wrong.

Reply:
But after it has been forcefully killed, is the PID going to help?

Reply:
Not really.